pkg/app/validate_app.go

```diff
@@ -99,7 +99,10 @@ func (v *Validator) Validate(request *v1beta1.AdmissionRequest) (bool, error) {
 	}
 
 	appAllowed, err := v.appValidator.ValidateApp(ctx, app)
-	if err != nil {
+	if validation.IsAppDependencyNotReady(err) {
+		v.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("skipping validation of app %#q in namespace %#q due to app dependency not ready yet", app.Name, app.Namespace))
+		return true, nil
+	} else if err != nil {
```

Comment:
@corest @kopiczko @tomahawk28 We will skip validating unique app CRs due to the version label.

https://github.com/giantswarm/app-admission-controller/blob/5f80e431202c06063a594f6786bc751ce8156fb4/pkg/app/validate_app.go#L96-L99

 But also matching the config not ready error to be extra safe in case we change that in future.

Reply:
Does that mean the app won't be validated at all in such case?

Reply:
Yes the app won't be validated due to the `0.0.0` value in the version label. But I wanted to also match the error in case we decide to change that later.

For now the admission controller is mainly to help customers creating app CRs via the control plane API.

Reply:
Ack.

Reply:
> Yes the app won't be validated due to the 0.0.0 value in the version label. 

WDYT checking the label first and if it is `0.0.0` then do not skipping validation?

``` 
if version == 0.0.0 {
 // validation
} else version < 3.0.0 {
 // skip validation
}
```

Reply:
The problem is then we can block our own deployments including app-admission-controller itself.

But if we want to validate unique app CRs we can add that.

Reply:
Please don't wait for me here. I think it's the best if you make the decision inside your team. I'm not too deep into the topic.